src/Debugger/Test/JsonTest.cs

```diff
@@ -41,12 +44,47 @@ public class JsonTest : IAsyncLifetime {
 
         [CompositeTest]
         [Category.R.Debugger]
-        [InlineData(@"'‘’“”'", @"""\u2018\u2019\u201c\u201d""")]
+        [InlineData("NULL", "null")]
+        [InlineData("NA", "null")]
+        [InlineData("NA_integer_", "null")]
+        [InlineData("NA_real_", "null")]
+        [InlineData("NA_character_", "null")]
+        [InlineData("TRUE[FALSE]", "null")]
+        [InlineData("0[FALSE]", "null")]
+        [InlineData("0L[FALSE]", "null")]
+        [InlineData("''[FALSE]", "null")]
+        [InlineData("TRUE", "true")]
+        [InlineData("FALSE", "false")]
+        [InlineData("0", "0")]
+        [InlineData("0L", "0")]
+        [InlineData(@"""text""", SameAsInput)]
+        [InlineData(@"""\r\n\t\f\""""", SameAsInput)]
+        [InlineData(@"""\a\v""", @"""\u0007\u000b""")]
+        [InlineData(@"""Ûñïçôδè ƭèжƭ: русский ελληνικά ქართული հայերեն हिन्दी ግዕዝ ᓀᐦᐃᔭᐍᐏᐣ 汉语/漢語 日本語 한국어/조선말 العَرَبِية‎ עברית""", SameAsInput)]
+        [InlineData(@"""‘’“”""", SameAsInput)]
+        [InlineData("list()", "[]")]
+        [InlineData("list(0, 's', NULL)", @"[0,""s"",null]")]
+        [InlineData("list(NA, 1, NA, 2, NA)", "[1,2]")]
+        [InlineData("structure(list(), names = ''[FALSE])", "{}")]
+        [InlineData("list(n = 0, s = 's', u = NULL)", @"{""n"":0,""s"":""s"",""u"":null}")]
+        [InlineData("list(n = 0, na = NA)", @"{""n"":0}")]
+        [InlineData("list(n = 0, na = NA)", @"{""n"":0}")]
+        [InlineData("as.environment(list())", "{}")]
+        [InlineData("as.environment(list(n = 0, s = 's', u = NULL))", @"{""n"":0,""s"":""s"",""u"":null}")]
+        [InlineData("as.environment(list(n = 0, na = NA))", @"{""n"":0}")]
+        [InlineData("as.environment(list(n = 0, na = NA))", @"{""n"":0}")]
+        [InlineData("list(as.environment(list(l = list())))", @"[{""l"":[]}]")]
```

Comment:
Not yet. It's a bit tricky to write a test for it, because invalid JSON is treated as a fatal error inside R-Host, which means that it immediately terminates the host process. I'm not sure what is the proper way to detect this in the test.